src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -749,4 +776,38 @@ public override IReadOnlyList<Type> GetParameterTypes(string methodName)
 
         return null;
     }
+
+    private static Activity? CreateActivity(IServiceProvider serviceProvider, string methodName)
+    {
+        if (serviceProvider.GetService<ActivitySource>() is ActivitySource activitySource
+            && activitySource.HasListeners())
+        {
+            var requestContext = Activity.Current?.Context;
+            // Get off the parent span.
+            // This is likely the Http Request span and we want Hub method invocations to not be collected under a long running span.
+            Activity.Current = null;
+
+            var activity = activitySource.CreateActivity($"{typeof(THub).Name}/{methodName}", ActivityKind.Server, parentId: null,
+                links: requestContext.HasValue ? [new ActivityLink(requestContext.Value)] : null);
+
+            if (activity is not null)
+            {
+                // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md#server-attributes
+                activity.AddTag("rpc.method", methodName);
+                activity.AddTag("rpc.system", "signalr");
```

Comment:
would it be possible to update otel semconv and add `signalr` constant to system enum? LMK if you need any help.